apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -100,32 +108,192 @@ export class ChangeAction extends BaseRushAction {
+    });
+
+    this._changeEmailParameter = this.defineStringParameter({
+      parameterLongName: '--email',
+      argumentName: 'EMAIL',
+      description: 'The email address to use in changefiles. If this parameter is not provided, the email address ' +
+        'will be detected or prompted for in intractive mode.'
+    });
+
+    this._bulkChangeParameter = this.defineFlagParameter({
+      parameterLongName: BULK_LONG_NAME,
+      description: 'If this flag is specified, apply the same change message and bump type to all changed projects. ' +
+        `The ${BULK_MESSAGE_LONG_NAME} and the ${BULK_BUMP_TYPE_LONG_NAME} parameters must be specified if the ` +
+        `${BULK_LONG_NAME} parameter is specified`
+    });
+
+    this._bulkChangeMessageParameter = this.defineStringParameter({
+      parameterLongName: BULK_MESSAGE_LONG_NAME,
+      argumentName: 'MESSAGE',
+      description: `The message to apply to all changed projects if the ${BULK_LONG_NAME} flag is provided.`
+    });
+
+    this._bulkChangeBumpTypeParameter = this.defineChoiceParameter({
+      parameterLongName: BULK_BUMP_TYPE_LONG_NAME,
+      alternatives: [...Object.keys(this._getBumpOptions()), ChangeType[ChangeType.none]],
+      description: `The bump type to apply to all changed projects if the ${BULK_LONG_NAME} flag is provided.`
+    });
   }
 
   public run(): Promise<void> {
     console.log(`The target branch is ${this._targetBranch}`);
     this._projectHostMap = this._generateHostMap();
 
     if (this._verifyParameter.value) {
+      const errors: string[] = ([
+        this._bulkChangeParameter,
+        this._bulkChangeMessageParameter,
+        this._bulkChangeBumpTypeParameter,
+        this._overwriteFlagParemter
+      ]).map((parameter) => {
+        return parameter.value
+        ? (
+          `The {${this._bulkChangeParameter.longName} parameter cannot be provided with the ` +
+            `${this._verifyParameter.longName} parameter`
+          )
+        : '';
+      }).filter((error) => error !== '');
+      if (errors.length > 0) {
+        errors.forEach((error) => console.error(error));
+        throw new AlreadyReportedError();
+      }
+
       this._verify();
       return Promise.resolve();
     }
 
-    this._sortedProjectList = this._getChangedPackageNames().sort();
-    if (this._sortedProjectList.length === 0) {
+    const sortedProjectList: string[] = this._getChangedPackageNames().sort();
+    if (sortedProjectList.length === 0) {
       this._logNoChangeFileRequired();
       this._warnUncommittedChanges();
       return Promise.resolve();
     }
 
-    this._prompt = inquirer.createPromptModule();
-    this._changeFileData = new Map<string, IChangeFile>();
-    this._changeComments = ChangeFiles.getChangeComments(this._getChangeFiles());
+    this._warnUncommittedChanges();
+
+    let changeFileDataPromise: Promise<Map<string, IChangeFile>>;
+    let allowOverwriteHandler: (filePath: string) => Promise<boolean>;
```

Comment:
This `allowOverwriteHandler` parameter is fairly strange.  Promises require us to treat blocks of program code as data objects, but that doesn't make it a good practice.  Can you just move the logic into `_writeChangeFile()`?  This code is private, so I don't see any problem with that function reading some variable like `this.shouldPromptForOvewrite` or whatever.

Reply:
BTW Node 6 isn't LTS any more.  Looks like if we drop support for Node 6, we should be safe to replace all our promises with `async`/`await`.  ;-)